server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregationBuilder.java

```diff
@@ -135,7 +136,22 @@ public boolean supportsSampling() {
     }
 
     @Override
-    public boolean supportsParallelCollection() {
+    public boolean supportsParallelCollection(ToLongFunction<String> fieldCardinalityResolver) {
+        // TODO unconditionally enable parallelism when ordered by key instead of count as precision is not affected
+        /*
+        if (InternalOrder.isKeyOrder(order)) {
+            return true;
+        }
+        */
+        if (script() == null
+            && (executionHint == null || executionHint.equals(TermsAggregatorFactory.ExecutionMode.GLOBAL_ORDINALS.toString()))) {
+            long cardinality = fieldCardinalityResolver.applyAsLong(name);
+            if (cardinality <= 0) {
+                return false;
+            }
+            BucketCountThresholds adjusted = TermsAggregatorFactory.adjustBucketCountThresholds(bucketCountThresholds, order);
```

Comment:
This makes a copy of `bucketCountThresholds`. But I think we need to update it as well? Otherwise the terms aggregation keep executing using the original `bucketCountThresholds`?

Reply:
I ran into problems when doing so, hence I went for this hack that only simulates the update but delays to a later time. I will try again once the build is green